### PR TITLE
Add pieces method, no tests or driver code

### PR DESCRIPTION
**Pull request by glnro**
@zencephalon  Built the pieces/board models. Let us know what your thoughts are.

### CODE REVIEW

**Comment by gp3gp3gp3** [comment]
@zencephalon @ChristineSchatz @GLNRO :(

File: board.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+class Board
+  include Mover
+  attr_reader :board, :key_array
+  def initialize
+    @board = {}
+    @key_array = []
+    board_hash
+  end
+
+  def board_maker
```
Comment (zencephalon) [inline board.rb:12]: I feel like this doesn't do what it says.

File: board.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    (0..7).each do |x|
+      (0..7).each do |y|
+        key_array << [x,y]
+      end
+    end
+  end
+
+  def board_hash
+    board_maker
```
Comment (zencephalon) [inline board.rb:21]: Weird for one method to pass data to the other through instance variables. If U want board_hash to use data from board_maker just have board_maker return an array. Like

`board_maker.each do |num|`

But name board_maker better.

File: board.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    key_array.each do |num|
+      key = num
```
Comment (zencephalon) [inline board.rb:23]: Why bother assigning these variables?

File: board.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      val = nil
+      board[key] = val
```
Comment (zencephalon) [inline board.rb:25]: Just do

`board[num] = nil`

File: board.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    end
+    @board
+  end
+
+  def to_s
+  #   # binding.pry
+    @board.to_s
+  end
+
+  def include?(array)
```
Comment (zencephalon) [inline board.rb:35]: Surely this parameter has more meaning than just `array`. Like `position`?

File: mover.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    x = pos[0]
+    y = pos[1]
+    vec_array.each do |vec|
+      while @board.include?([x, y])
+        x += vec[0]
+        y += vec[1]
+        if @board.include?([x, y])
+          ary << [x,y]
+        else
+          x = pos[0]
```
Comment (zencephalon) [inline mover.rb:13]: `x, y = pos` let Ruby do the work for U.

File: mover.rb [hunk 7]
```diff
@@ -0,0 +1,21 @@
+module Mover
+  def move(pos, vec_array)
+    ary = []
+    x = pos[0]
```
Comment (zencephalon) [inline mover.rb:4]: `x, y = pos`

File: pieces.rb [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+      ary << [x, y + 2]
+    end
+    if @board[x + 1, y + 1].color == :black
+      ary << [x + 1, y + 1]
+    end
+  elsif @board[x - 1, y + 1].color == :black
+    ary << [x - 1, y + 1]
+  end
+  return ary
+    if color == :black
```
Comment (zencephalon) [inline pieces.rb:61]: Hmm... all of this seems vaguely similar to the code for `color == :white`. Any way U could DRY up the two clauses?

**Comment by zencephalon** [comment]
@GLNRO @gp3gp3gp3 @ChristineSchatz ILUVU, nice work on your piece logic here. I've got some comments for U.

**Pull request closed without merging**